compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -708,6 +725,26 @@ static void PrintStub(
               "$output_type$ $lower_method_name$($input_type$ request)");
         }
         break;
+      case BLOCKING_V2_CALL:
+        if (client_streaming) { // Both Bidi and Client Streaming
+          p->Print(
+              *vars,
+              "$BlockingClientCall$<$input_type$, $output_type$>\n"
+               "    $lower_method_name$()");
+        } else if (server_streaming) {
+          // Server streaming
+          p->Print(
+              *vars,
+              "$BlockingClientCall$<?, $output_type$>\n"
+              "    $lower_method_name$($input_type$ request) throws java.lang.InterruptedException,\n"
```

Comment:
We should discuss whether we remove InterruptedException here. (We know it won't throw.)

Reply:
This is still an open item.

Reply:
ClientCallImpl skips onReady() callbacks for unary and server-streaming:
https://github.com/grpc/grpc-java/blob/7b29111cd0a4bd605254b0b93bb45da590e1e245/core/src/main/java/io/grpc/internal/ClientCallImpl.java#L740-L743

So we can't block here, as we may never get the onReady callback. The test didn't notice this because it uses in-process and in-process connects immediately.